lib/improver/tests/nowcasting/accumulation/test_Accumulation.py

```diff
@@ -0,0 +1,413 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2019 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Unit tests for the nowcasting.Accumulation plugin """
+
+import datetime
+import unittest
+import numpy as np
+
+import iris
+from iris.tests import IrisTest
+
+from improver.utilities.warnings_handler import ManageWarnings
+from improver.nowcasting.accumulation import Accumulation
+from improver.tests.set_up_test_cubes import set_up_variable_cube
+
+
+class rate_cube_set_up(IrisTest):
+    """Set up a sequence of precipitation rates cubes for use in testing the
+    accumulation plugin functionality."""
+
+    def setUp(self):
+        """Set up test cubes."""
+        ncells = 10
+        # Rates equivalent to 5.4 and 1.8 mm/hr
+        rates = np.ones((ncells)) * 5.4
+        rates[0:ncells//2] = 1.8
+        rates = rates / 3600.
+
+        datalist = []
+        for i in range(ncells):
+            data = np.vstack([rates] * 4)
+            data = np.roll(data, i, axis=1)
```

Comment:
Can you add a comment on what this data looks like please.  I can see it has values of 1.8 and 5.4 mm/h, but not what shape it is, where the mask is or where (roughly) different values are located.

For the value tests, I think you need different rate values for different time steps so that you can demonstrate half-weighting at the end points (or not, as the case may be).

Reply:
The rate at any given point is changing in time (not continuously, but discretely). The accumulations are calculated at each grid point, so considering the figure below it can be seen that the rate for a point in the top right changes from a high to low rate, whilst a point in the top left sees a low rate stop after a time. The whole scenario is designed to be an intense shower moving off to the right with some lighter precip falling as it moves away before becoming dry.

![image](https://user-images.githubusercontent.com/28042362/60946760-ed5d4000-a2e6-11e9-9d7f-e21ed886cb27.png)

So there are time steps for a given point that will see a rate change. Moving to a centre weighted calculation of accumulations will affect various points. Consider those points in the far upper left that previously applied the low precip rate over the 1 minute period between the first two frames, these now take an average of this low precip rate and a zero rate which is present at T+1 minute. This results in an accumulation of half that which was previously calculated.

I am implementing the centre weighted calculation of accumulations as you state that this is more standard, but I don't believe the tests need to change beyond adapting the values to reflect the new results.

Reply:
Added.

Reply:
Thanks.